src/App.jsx

```diff
@@ -1,33 +1,54 @@
+import { useState } from 'react';
+
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-export const App = () => (
-  <div className="page">
-    <div className="page-content">
-      <div className="box">
-        <div className="field">
-          {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-          <label htmlFor="search-query" className="label">
-            Search movie
-          </label>
-
-          <div className="control">
-            <input
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+const filterBy = (query, setMovies) => {
+  const newMovies = moviesFromServer.filter(
+    movie => movie.title.toLowerCase().includes(query.trim().toLowerCase())
+      || movie.description.toLowerCase().includes(query.trim().toLowerCase()),
+  );
+
+  setMovies(newMovies);
+};
+
+export const App = () => {
+  const [visibleMovies, setMovies] = useState(moviesFromServer);
+
+  return (
+    <div className="page">
+      <div className="page-content">
+        <div className="box">
+          <div className="field">
+            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+            <label htmlFor="search-query" className="label">
+              Search movie
+            </label>
+
+            <div className="control">
+              <input
+                type="text"
+                id="search-query"
+                className="input"
+                placeholder="Type search word"
+                onChange={(event) => {
+                  filterBy(
+                    event.target.value,
+                    setMovies,
+                  );
+                }}
+              />
```

Comment:
it is bad practise to make anonymous arrow function in jsx.
suggest to make function in component and pass referense here
```
const handleChange = (event) => {
  filterBy(
    event.target.value,
    setMovies,
  );
}
```
```suggestion
                onChange={handleChange}
              />
```